pkgs/tools/virtualization/nixos-nspawn/default.nix

```
@@ -0,0 +1,43 @@
+{ stdenv, lib, python3, nixosTests }:
+
+stdenv.mkDerivation rec {
+  name = "nixos-nspawn";
+  src = lib.cleanSourceWith {
+    src = ./.;
+    filter = path: type:
+      type == "regular" && ! (lib.hasSuffix "default.nix" path);
+  };
+
+  buildInputs = [ python3 python3.pkgs.wrapPython ];
+
+  postPatch = ''
+    sed -i -e '1,2d' ./nixos-nspawn.py
+    sed -i -e '1s;^;#!${python3}/bin/python3;' ./nixos-nspawn.py
```

How about nixos-nspawn begins like this

```
#!/usr/bin/env nix-shell
#!nix-shell -i python3 -p "python3.withPackages(ps: with ps; [ rich ])"

#!/usr/bin/env python3
```

and we strip the first 3 lines? I think that is easier to understand than injecting the shebang. Took me a little while at least.

```suggestion
    # remove nix-shell shebang
    sed -i -e '1,3d' ./nixos-nspawn.py
```